diff --git a/site_scons/SGppConfigure.py b/site_scons/SGppConfigure.py
index e5f5493..01525fa 100644
--- a/site_scons/SGppConfigure.py
+++ b/site_scons/SGppConfigure.py
@@ -87,8 +87,8 @@ def doConfigure(env, moduleFolders, languageWrapperFolders):
     # -q is the relevant flag to make it use integrated assembler
     if config.env["COMPILER"] == "gcc":
       config.env.AppendUnique(CPPFLAGS=["-Wa,-q"])
-    config.env.AppendUnique(CPPPATH="/usr/local/include")
-    config.env.AppendUnique(LIBPATH="/usr/local/lib")
+    #config.env.AppendUnique(CPPPATH="/usr/local/include")
+    #config.env.AppendUnique(LIBPATH="/usr/local/lib")
     config.env["SHLIBSUFFIX"] = ".dylib"
   elif config.env["PLATFORM"] == "cygwin":
     # required to find the static libraries compiled before the shared libraries
